Error on GetNativeManifest when P2P to a multitargeting project with ReferenceOutputAssembly=false

Small repro: [sln-multitargeting.zip](https://github.com/Microsoft/msbuild/files/1674451/sln-multitargeting.zip)

After #2867 (so for current VS 15.6 previews), that project will fail to build with 

```
s:\work\sln-multitargeting\multitargeting\multitargeting.csproj : error MSB4057: The target "GetNativeManifest" does not exist in the project.
```

This worked before because the reference went to the inner build of a correct TF.

This came up internally when the calling project built a VSIX package and targeted `net461`, and wanted to pack the results of another project that targeted `net461;netcoreapp2.0`:

> So we reference the multitargeted project from the VSIX and set `ReferenceOutputAssembly=false` and also set `IncludeOutputGroupsInVSIX` to a target on the multitargeted project that performs a publish of just `netcoreapp2.0` and populates the output of the publish (plus some other dependencies) to an outputgroup that’s included in the VSIX.
> 
> This used to work fine but now with the above mentioned change, since TF checks are being skipped, I assume `ResolveProjectReferences` is trying to call the outer build of the multitargeted project's `GetNativeManifest`, which ofcourse doesn’t exist. I added
> ```xml
> <SetTargetFramework>TargetFramework=netcoreapp2.0</SetTargetFramework>
> ```
> to the projectreference in the vsix project and then it builds fine. 

That's a good workaround (and captures intent better than the sort-of-mysterious TF compat check, which was actually pointing to the net461 TF and then hopping through the custom target to the right one), but we shouldn't break in this way.

This could be fixed by doing #2877 for the `GetNativeManifest` target (at least).

Thanks for the report, @srivatsn.

--- a/src/Tasks/Microsoft.Common.CurrentVersion.targets
+++ b/src/Tasks/Microsoft.Common.CurrentVersion.targets
@@ -1545,15 +1545,9 @@ Copyright (C) Microsoft Corporation. All rights reserved.
 
       We give this treatment to .vcxproj by default since no .vcxproj can target more
       than one framework.
-
-      Likewise if the dependency is for build ordering instead of an assembly reference
-      (ReferenceOutputAssembly=false), skip the checks since we can't know what TF
-      the output would need to be compatible with.
    -->
    <ItemGroup>
-      <_MSBuildProjectReferenceExistent
-        Condition="'%(_MSBuildProjectReferenceExistent.SkipGetTargetFrameworkProperties)' == '' and
-                   ('%(Extension)' == '.vcxproj' or '%(ReferenceOutputAssembly)' == 'false')">
+      <_MSBuildProjectReferenceExistent Condition="'%(_MSBuildProjectReferenceExistent.SkipGetTargetFrameworkProperties)' == '' and '%(Extension)' == '.vcxproj'">
         <SkipGetTargetFrameworkProperties>true</SkipGetTargetFrameworkProperties>
       </_MSBuildProjectReferenceExistent>
    </ItemGroup>

